File: Library/demo/clustering/object_detection.R
imports "clustering" from "MLkit";

require(graphics2D);

options(strict = FALSE);
setwd(@dir);

multishapes = read.csv("./multishapes.csv");

print(multishapes, max.print = 13);

# detect object from umap data
objects = graphics2D::pointVector(multishapes[, "x"], multishapes[, "y"]) |> dbscan_objects();
objects[objects == "-1"] = "noise";
objects = ifelse(objects == "noise", objects, `object_${objects}`);

# show object detection result
bitmap(file = "./object_detection.png") {
	plot(multishapes[, "x"], multishapes[, "y"], 
		class     = objects, 
		grid.fill = "white",
		padding   = "padding: 125px 300px 200px 200px;",
		colorSet  = "paper"
	);
}

File: test/demo/clustering/object_detection.py
# load R# library for run dbscan
from MLkit import clustering 
# load R# library for plots
import graphics2D

setwd(@dir)

multishapes = read.csv("./multishapes.csv")
x, y        = (multishapes[, "x"], multishapes[, "y"])

print(`contains ${x.length()} data points.`)
print(multishapes, max.print = 13)

# detect object from umap data
objects = `object_${graphics2D::pointVector(x, y).dbscan_objects()}`
# show object detection result
margin  = "padding: 125px 300px 200px 200px;"
plt     = plot(x, y, class = objects, grid.fill = "white", padding = margin)

bitmap(plt, file = "./object_detection_py.png")